django/librarian/utils/process_engine.py

```diff
@@ -180,22 +200,24 @@ def extract_markdown(
     return md, md_chunks
 
 
-def pdf_to_markdown(content):
-    html = _pdf_to_html_using_azure(content)
+def pdf_to_markdown_azure_layout(content):
+    html = _pdf_to_html_azure_layout(content)
     return _convert_html_to_markdown(html)
 
 
-def fast_pdf_to_text(content):
-    # Note: This method is faster than using Azure Form Recognizer
-    # Expected it to work well for more generic scenarios but not for scanned PDFs, images, and handwritten text
+def pdf_to_text_pdfium(content):
+    # Fast and cheap, but no OCR or layout analysis
     import pypdfium2 as pdfium
 
-    pdf = pdfium.PdfDocument(content)
     text = ""
+    pdf = pdfium.PdfDocument(content)
     for i, page in enumerate(pdf):
+        text_page = page.get_textpage()
         text += f"<page_{i+1}>\n"
-        text += page.get_textpage().get_text_range() + "\n"
+        text += text_page.get_text_range() + "\n"
         text += f"</page_{i+1}>\n"
+        text_page.close()
+    pdf.close()
```

Comment:
The explicit `.close()` is necessary with pypdfium to avoid memory leaks (or at least, warnings about memory leaks).